### PR TITLE
Pull request conflict files detection

### LINKED ISSUE #887: Show conflicted files on pull request ui.
Currently when a PR is creating, it will return 500 if there are some conflicts. It's not very friendly UI.

### PR DESCRIPTION
**Pull request by lunny**
This pull request will add a new column on table `pull_request` named `conflicted_files` that will store less than 10 conflicted file names which will be filled on `testPatch` method. Will fix #887

### CODE REVIEW

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -871,6 +871,7 @@ pulls.has_merged = The pull request has been merged.
 pulls.title_wip_desc = `<a href="#">Start the title with <strong>%s</strong></a> to prevent the pull request from being merged accidentally.`
 pulls.cannot_merge_work_in_progress = This pull request is marked as a work in progress. Remove the <strong>%s</strong> prefix from the title when it's ready
 pulls.data_broken = This pull request is broken due to missing fork information.
+pulls.files_conflicted = This pull request has some files conflicted with target branch.
```
Comment (lafriks) [inline options/locale/locale_en-US.ini]:
```suggestion
pulls.files_conflicted = This pull request has changes conflicting with the target branch.
```

File: models/pull.go [hunk 2]
```diff
@@ -1396,6 +1416,11 @@ func (pr *PullRequest) IsWorkInProgress() bool {
 	return false
 }
 
+// IsFilesConflicted determine if the  Pull Request has files conflicted.
```
Comment (adelowo) [inline models/pull.go]: I would have ___determines___ here instead

Comment (adelowo) [reply models/pull.go]: Going with @lafriks's suggestion with the translation

```suggestion

// IsFilesConflicted determines if the Pull Request has changes conflicting with the target branch.

```

**Review by adelowo — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by bederke** [comment]
What about this feature? According to https://blog.gitea.io/2019/04/gitea-1.8.0-is-released/, it is released already, but I cannot find it in Gitea.

**Comment by lunny** [comment]
@bederke When you create a pull request which has conflict files. You can find them on pull request view interface.

**Comment by mschoettle** [comment]
I have a PR that shows a conflicted file but when I merge it locally there is no conflict. Is this a bug in the conflicted files detection?

**Comment by jolheiser** [comment]
@mschoettle Please open a new issue if you are experiencing a problem. 🙂 

Issues on merged PRs tend to get lost fairly quickly.

**Pull request merged**